src/data/java/com/github/minecraftschurli/arsmagicalegacy/data/AMEnglishLanguageProvider.java

```diff
@@ -0,0 +1,196 @@
+     */
+    private void blockIdTranslation(Supplier<? extends Block> block) {
+        addBlock(block, idToTranslation(block.get().getRegistryName().getPath()));
+    }
+
+    /**
+     * Adds a block translation.
+     *
+     * @param block       The block to generate the translation for.
+     * @param translation The translation to apply.
+     */
+    private void blockTranslation(Supplier<? extends Block> block, String translation) {
+        addBlock(block, translation);
+    }
+
+    /**
+     * Adds an item translation that matches the item id.
+     *
+     * @param item The item to generate the translation for.
+     */
+    private void itemIdTranslation(Supplier<? extends Item> item) {
+        addItem(item, idToTranslation(item.get().getRegistryName().getPath()));
+    }
+
+    /**
+     * Adds an item translation.
+     *
+     * @param item        The item to generate the translation for.
+     * @param translation The translation to apply.
+     */
+    private void itemTranslation(Supplier<? extends Item> item, String translation) {
+        addItem(item, translation);
+    }
+
+    /**
+     * Adds an effect translation that matches the effect id.
+     *
+     * @param effect The effect to generate the translation for.
+     */
+    private void effectIdTranslation(Supplier<? extends MobEffect> effect) {
+        addEffect(effect, effect.get().getRegistryName().getPath());
+    }
+
+    /**
+     * Adds an enchantment translation that matches the enchantment id.
+     *
+     * @param enchantment The enchantment to generate the translation for.
+     */
+    private void enchantmentIdTranslation(Supplier<? extends Enchantment> enchantment) {
+        addEnchantment(enchantment, enchantment.get().getRegistryName().getPath());
+    }
+
+    /**
+     * Adds an entity translation that matches the entity id.
+     *
+     * @param entity The entity to generate the translation for.
+     */
+    private void entityIdTranslation(Supplier<? extends EntityType<?>> entity) {
+        addEntityType(entity, entity.get().getRegistryName().getPath());
+    }
+
+    /**
+     * Adds a creative tab translation that matches the effect id.
+     *
+     * @param translation The creative tab to generate the translation for.
+     */
+    private void creativeTabTranslation(CreativeModeTab tab, String translation) {
+        add("itemGroup." + tab.getRecipeFolderName(), translation);
+    }
+
+    /**
+     * Adds an advancement translation.
+     *
+     * @param advancement The advancement id.
+     * @param title       The advancement title.
+     * @param description The advancement description.
+     */
+    private void advancementTranslation(String advancement, String title, String description) {
+        add("advancements." + ArsMagicaAPI.MOD_ID + "." + advancement + ".title", title);
+        add("advancements." + ArsMagicaAPI.MOD_ID + "." + advancement + ".description", description);
+    }
```

Comment:
use Util.makeDescriptionId() and ResourceLocation 
```suggestion
    private void advancementTranslation(ResourceLocation advancement, String title, String description) {
        add(Util.makeDescriptionId("advancement", advancement) + ".title", title);
        add(Util.makeDescriptionId("advancement", advancement) + ".description", description);
    }
```